src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -968,12 +968,16 @@ namespace smt::noodler {
                 auto [noodler_lengths, precision] = dec_proc->get_lengths();
                 lengths = len_node_to_z3_formula(noodler_lengths);
                 lbool is_lengths_sat = check_len_sat(lengths);
-
-                // we assume that precision != LenNodePrecision::OVERAPPROX
                 
                 if (is_lengths_sat == l_true) {
                     STRACE("str", tout << "len sat " << mk_pp(lengths, m) << std::endl;);
-                    last_run_was_sat = true;
```

Comment:
Was this lost in rebasing?

Reply:
Probably. It should be there?

Reply:
Yeah, `block_curr_len` should not be called there anymore (as it was than only to keep it in loop protection, which is not needed).

Reply:
Done